source/slang/slang-lower-to-ir.cpp

```diff
@@ -5387,65 +5406,78 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         auto subContext = nestedContext.getContext();
 
         // Emit any generics that should wrap the actual type.
-        emitOuterGenerics(subContext, decl, decl);
+        auto outerGeneric = emitOuterGenerics(subContext, decl, decl);
 
-        IRStructType* irStruct = subBuilder->createStructType();
-        addNameHint(context, irStruct, decl);
-        addLinkageDecoration(context, irStruct, decl);
+        IRInst* resultType = nullptr;
+        if (as<AssocTypeDecl>(decl))
+        {
+            resultType = subBuilder->createAssociatedType();
+        }
```

Comment:
Yeah, this bit seems broken to me. We shouldn't be emitting a global `IRInst` to represent each associated type and giving it a mangled name (and thus linkage), when the only context that needs to refer to it will be the `IRInterfaceType` that declares it. These `IRAssociatedType` insts really seem like they should be scoped as narrowly as possible, and it seems like the lowering for the interface type could easily just create all of them as a pre-process before it dives into emitting the requirement entries.

Reply:
Fixed.